pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+* The scrollbar is "disabled" (not necessary anymore).
+* The whole page is scrolled back to top and we can see the header again.
+
+As you can see, when the chart is displayed, the mouse cursor is now above some text.
+
+## The absolute position solution
+
+Before CSS grid, I would have solved this problem with the [position CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/position).
+
+First, I would use `visibility: hidden` instead of `display: none` to hide the "inactive" panel:
+
+![UI Component displaying some help](./visibility-hidden-full-height.png)
+
+As you can see, with `visibility: hidden`, the "inactive" panel is hidden but the component computes its own size as if both panels were there.
+Now, we have a component with a stable height, based on the heights of `.chart` and `.info`.
+
+Then, I would use `position: absolute` on the `.chart`.
+Doing this will [remove the element from normal flow](https://developer.mozilla.org/en-US/docs/Web/CSS/position#absolute).
+This means the component will compute its own size as if `.chart` was not there.
+Therefore, the component size would mostly depend on the size of `.info`, which is exactly what we want.
+
+The only step left is to give `.chart` exactly the same position and size as `.info`.
+Using `position: absolute` means `.chart` will be positioned *"relative to its closest positioned ancestor"*.
+It means, an ancestor with a `position` different from the default value (`static`).
+Most of the time, we use `position: relative` on the parent to trigger this.
+In this situation, we would need an additional `.wrapper` around both panels like this:
+
+```html
+<div class="chart-component">
+  <div class="title">
+    HTTP response codes <button>toggle chart/info</button>
+  </div>
+  <div class="wrapper">
+    <div class="chart">
+      <!-- chart here -->
+    </div>
+    <div class="info">
+      <!-- short text here -->
+    </div>
+  </div>
+</div>
+```
+
+Now that we have this HTML, the CSS solution would look like this:
+
+```css
+.wrapper {
+  /* .wrapper is the closest positioned ancestor of .chart */
+  position: relative;
+}
+
+.chart {
+  /* .chart is removed from normal flow, therefore
+     .wrapper has the exact same size (and position) as .info
+     because .info is the only child left in the normal flow
+     .chart is positioned relatively to .wrapper */
+  position: absolute;
+  /* same position as .wrapper (thus same position as .info) */
+  left: 0;
+  top: 0;
+  /* same size as .wrapper (thus same size as .info) */
+  height: 100%;
+  width: 100%;
+}
+```
+
+If you're still struggling with how the position CSS property works, this technique can be scary but don't worry:
+
+* It's OK. You're not alone!
+* It gets better with time and practice... 
+
+This solution helped us to achieve our goal.
+The size of our component will now always depend on the size of `.info`, even when this panel is hidden and the chart is displayed.
+
+😎 No more layout shift!
```

Comment:
Between my last comment and this one, I would write something like "The trick is to remove the _smallest_ block from the flow. That way, `.wrapper` will be the size of the bigger one."
I have no idea where the best place is, though.

That would help you explain why this solution is not satisfying. (A rule of thumb I learnt (in french for the alliteration) : « Accélérer l'accès au sens »)
I know you already explain that later (at the end of your grid solution). But that's way too late.

Reply:
I agree. The article needs to emphasise the drawback of the absolute positioning solution. Without the explanation, why should the reader bother to learn a new technique if the old ways perform well?

Reply:
Thanks, that's exactly the kind of feedback I need.
When I wrote that, I was hesistating and I guess the rythm of an article is not the same as a conference talk.

Reply:
I'll also emphasize the fact that walking into position: absolute territory is dangerous :laughing: